### ISSUE TITLE
Remove deprecated Core APIs

### ISSUE THREAD

**Issue by jodal** [mopidy/mopidy#1461]
Our Core API is starting to become quite cluttered with old deprecated APIs that we need to remove to make the API easier to use correctly. In additon, removing the old `@property` APIs will lead to some performance improvements, given the way Pykka proxies currently accesses the properties when inspecting the actor's interface.

Before we can remove the deprecated core APIs, we need to stop using them. Luckily, none of the WebSocket API based clients have ever had access to the deprecated properties, reducing the scope of this a great deal.
- [x] ~Update Mopidy tests to not use deprecated APIs. See #1083.~
- [x] Review Mopidy itself for deprecated API use. For example by removing the deprecated Core APIs, then running automated and manual tests.
- [x] ~Review all Python-based Mopidy Core API users. See https://github.com/jodal/mopidy-core-api-users for an easy way to search across the source code of many extensions.~

Extensions that needs to be updated:
- [x] Mopidy-MPRIS (mopidy/mopidy-mpris#13)
- [x] ~...more extensions identified by the review mentioned above...~

Remove deprecated APIs:
- `Core`
  - [x] `uri_schemes` property
  - [x] `version` property
- `LibraryController`
  - [x] `find_exact()` method
  - [x] `lookup()` method's `uri` argument
  - [x] `search()` method's empty query returns everything behavior
  - [x] `search()` method's `kwargs` argument
- `PlaybackController`
  - [x] `current_tl_track` property
  -  [x] `current_track` property
  - [x] `state` property
  - [x] `time_position` property
  - [x] `get_volume()` method
  - [x] `set_volume()` method
  - [x] `volume` property
  - [x] `get_mute()` method
  - [x] `set_mute()` method
  - [x] `mute` property
- `PlaylistController`
  - [x] `get_playlists()` method
  - [x] `playlists` property
  - [x] `filter()` method
- `TracklistController`
  - [x] `tl_tracks` property
  - [x] `tracks` property
  - [x] `length` property
  - [x] `version` property
  - [x] `consume` property
  - [x] `random` property
  - [x] `repeat` property
  - [x] `single` property
  - [ ] `add()` method's `tracks` argument
  - [x] `add()` method's `uri` argument
  - [x] `filter()` method's `kwargs` argument
  - [x] `remove()` method's `kwargs` argument

**Comment by jodal** [comment]
We should ignore the items about other extensions above. Extensions will have to cross this bridge themselves.

What remains to close this issue is to:

- [ ] Update `tests/core/test_playback.py` to not use `core.tracklist.add()`'s `tracks` argument. (This is also tracked as #1083).
- [ ] Remove `core.tracklist.add()`'s `tracks` argument.

**Comment by jodal** [comment]
I've deferred removing the `tracks` arg to `core.tracklist.add()` to Mopidy 4.0, ref. #1855.
